### PR TITLE
Feature: App running with `npm install`

### LINKED ISSUE #1246: App must run without using `npm install --legacy-peer-deps`
**Is your feature request related to a problem? Please describe.**

1. The app depends on legacy dependencies
1. The app must be upgraded to not use `--legacy-peer-deps`

**Describe the solution you'd like**
1. Upgrade app packages to not require the `--legacy-peer-deps`  qualifier when running `npm install`
1. Update the INSTALLATION.md file with the new installation process just using `npm install`
1. The  `npm install` must not redirect to `npm install --legacy-peer-deps` in any hidden way. 
1. `npm install --legacy-peer-deps` must not be invoked in any way behind the scenes to install the latest required packages
1. All existing tests must pass
1. No functionality must be removed

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- See above

**Additional context**
- N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by git-init-priyanshu**
**What kind of change does this PR introduce?**
feature

**Issue Number:**
Fixes #1246 

**Did you add tests for your changes?**
No

**Snapshots/Videos:**
Before:
image
After:
image

**If relevant, did you update the documentation?**
Yes

**Summary**
Right now, we cannot use `npm install` as it throws a bunch of  errors, and can only be solved by using the `--legacy-peer-deps` flag. This PR fixes those peer dependency issues.

**Does this PR introduce a breaking change?**
No.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by aashimawadhwa** [comment]
can you share a video of how `npm i` is working fine now? @git-init-priyanshu

**Comment by git-init-priyanshu** [comment]
@aashimawadhwa 
https://drive.google.com/drive/folders/1r3E-92VxvLBiFpF4kpxpjvrON-yAe7Mj

**Comment by palisadoes** [comment]
Was the `enzyme-adapter-react-16` package ever used in the code base? You've only edited the package listing.

**Comment by git-init-priyanshu** [comment]
> Was the `enzyme-adapter-react-16` package ever used in the code base? You've only edited the package listing.

No sir, it was not used in the code. The solution was to install `@wojtekmaj/enzyme-adapter-react-17` package which is compatible with react 17. It was already installed and correctly used.

**Comment by git-init-priyanshu** [comment]
Right now I am trying to solve peer depedency warnings. And I also need to change the INSTALLATION.md for the node version.

**Comment by git-init-priyanshu** [comment]
Hey @aashimawadhwa can you check this PR now?

**Comment by palisadoes** [comment]
@noman2002 @anwersayeed @aashimawadhwa please review

File: src/screens/OrgPost/OrgPost.tsx [hunk 1]
```diff
@@ -227,7 +227,7 @@ function orgPost(): JSX.Element {
                     </Dropdown.Toggle>
                     <Dropdown.Menu>
                       <Dropdown.Item
-                        value="searchText"
+                        // value="searchText"
```
Comment (noman2002) [inline src/screens/OrgPost/OrgPost.tsx:230]: Why this is commented ?

File: src/screens/OrgPost/OrgPost.tsx [hunk 2]
```diff
@@ -237,7 +237,7 @@ function orgPost(): JSX.Element {
                         {t('Text')}
                       </Dropdown.Item>
                       <Dropdown.Item
-                        value="searchTitle"
+                        // value="searchTitle"
```
Comment (rishav-jha-mech) [inline src/screens/OrgPost/OrgPost.tsx:240]: This PR is not related to OrgPost, so please revert any change you made to this file

**Comment by git-init-priyanshu** [comment]
hey @noman2002 @rishav-jha-mech, Sorry for my late reply, I didn't receive the notification.
I commented those two lines because I was not sure what to do as they were throwing error "Property 'value' does not exist ...". I also checked all the reference of "Dropdown.Item" in the whole codebase and the value property was only defined here. And I also checked the API of "Dropdown.Item" and there was also no mention of value property. Let me know if this is a mistake I will correct it.

**Comment by git-init-priyanshu** [comment]
hey @rishav-jha-mech @noman2002, I have fixed the issue and updated the INSTALLATION.md file.

**Comment by rishav-jha-mech** [comment]
> hey @noman2002 @rishav-jha-mech, Sorry for my late reply, I didn't receive the notification. I commented those two lines because I was not sure what to do as they were throwing error "Property 'value' does not exist ...". I also checked all the reference of "Dropdown.Item" in the whole codebase and the value property was only defined here. And I also checked the API of "Dropdown.Item" and there was also no mention of value property. Let me know if this is a mistake I will correct it.

It maybe a eslint error, make sure you have eslint extension installed in your VS Code

File: package.json [hunk 3]
```diff
@@ -47,7 +45,6 @@
     "jest-docblock": "^27.4.0",
     "js-cookie": "^3.0.1",
     "markdown-toc": "^1.2.0",
-    "node-sass": "^9.0.0",
```
Comment (rishav-jha-mech) [inline package.json:50]: Don't remove this package, it is used to compile customised sass for our project

**Comment by rishav-jha-mech** [comment]
@git-init-priyanshu after all these changes are the packages getting installed without `--legacy-peer-deps` with npm instal ?

**Comment by git-init-priyanshu** [comment]
> @git-init-priyanshu after all these changes are the packages getting installed without `--legacy-peer-deps` with npm instal ?

yes. Here is the video proof. https://drive.google.com/drive/folders/1j66OogD9dgTzPOLHCLYjN_KAP_CCj4sO

File: package.json (hunk 3 above)
Comment (git-init-priyanshu) [reply package.json:50]: Ok. So should I revert the changes? It was causing some peer-dependency issues.

Comment (rishav-jha-mech) [reply package.json:50]: We need this module, we can't just remove our project's dependency if its causing peer-dependency issues, it would be better if you can find a way around to fix this

**Comment by git-init-priyanshu** [comment]
> @git-init-priyanshu I have seen the uploaded video, let me know if having node-sass module is causing any issue, if not then we will merge your PR

Right now everything is working fine. But I found some sass files under `src/assets/scss` , So if these files need to be compiled then I think we do need the node-sass.

**Comment by rishav-jha-mech** [comment]
@git-init-priyanshu please put node-sass back into package.json, and let us know if the npm install is working or not

**Comment by git-init-priyanshu** [comment]
> @git-init-priyanshu please put node-sass back into package.json, and let us know if the npm install is working or not

I reverted the changes. And `npm install` is working fine.

**Review by rishav-jha-mech — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]
